nix/haskell.nix

```diff
@@ -147,10 +147,6 @@ haskell-nix: haskell-nix.cabalProject' [
         ] ++ lib.filter
           (drv: lib.isDerivation drv && drv.name != "regenerate-materialized-nix")
           (lib.attrValues haskell-build-tools));
-
-        CARDANO_NODE_CONFIGS = pkgs.cardano-node-deployments;
```

Comment:
Had broken with
```
error: The option `shell.CARDANO_NODE_CONFIGS' does not exist. Definition values:
       - In `<unknown-file>': <derivation /nix/store/fh48mcfjlixkh7xb7s76q2d011yplxjw-cardano-node-deployments.drv>
(use '--show-trace' to show detailed location information)
warning: Git tree '/Users/anviking/IOHK/cardano-wallet' is dirty
```

I believe removing this line is fine as it is unneeded.

`CARDANO_NODE_CONFIGS` is independently set elsewhere in `test/e2e/shell.nix` and `packages.cardano-wallet.components.benchmarks.restore`.